### PR TITLE
Keep every file in the index when built without CGO

### LINKED ISSUE #9: Released binaries ship an empty repo map: CGO_ENABLED=0 makes the index drop every file
Tracked in [ROADMAP.md](https://github.com/BaryoDev/Baryo.CLI/blob/main/ROADMAP.md) → P0 (Correctness & Trust). Context for constrained deployments: [DEPLOYMENT.md](https://github.com/BaryoDev/Baryo.CLI/blob/main/DEPLOYMENT.md) → "Build and packaging: the CGO question".

## Summary

Every released binary indexes **zero files** and produces an **empty repo map**. This is not a graceful degradation to "no symbols" — the index ends up completely empty.

`.goreleaser.yaml:16` sets `CGO_ENABLED=0`. Tree-sitter requires CGO, so `internal/index/parser_nocgo.go` is compiled in. Its `ParseFile` returns valid file metadata *and* a non-nil error:

```go
return &FileSymbols{
    Path:    path,
    ModTime: modTime,
    Size:    size,
}, fmt.Errorf("tree-sitter parsing requires CGO")
```

`Index.parseOne` propagates that error (`internal/index/index.go:136-139`), and both `Build` (`index.go:52-55`) and `Update` (`index.go:112-115`) do `if err != nil { continue }` — so every file is skipped and `idx.files` stays empty.

## Measured

Probe against this repository, identical code, only `CGO_ENABLED` differing:

| Build | Files discovered | Files indexed | `RepoMap(2000)` |
|---|---|---|---|
| `CGO_ENABLED=1` | 175 | 175 | 5780 bytes, symbols extracted |
| `CGO_ENABLED=0` | 175 | **0** | **0 bytes** |

`ParseFile` on `main.go` returns 7 symbols under CGO, and `err=tree-sitter parsing requires CGO` with 0 symbols without.

## Impact

- Affects **every distribution channel on every platform** — Homebrew, Scoop, `install.sh`, all `goreleaser` artifacts — not just ARM. Only users who `go build` from source with CGO enabled have a working repo map.
- The model runs with no repo structure at all. This is worst for small local models, where an accurate repo map is the highest-value use of a limited context window.
- README documents tree-sitter repo mapping across 8 languages as a headline feature; the shipped artifact has none of it.

## Why CI did not catch it

`.github/workflows/ci.yml` runs tests under both `CGO_ENABLED=0` and `CGO_ENABLED=1`, but:

- it only ever **builds the binary** with `CGO_ENABLED=0` (`ci.yml:38-39`), and
- the CGO test pass does not assert that indexing yields a non-empty index, so the no-CGO path passing with zero indexed files looks green.

## Proposed fix

**1. Stop the bleeding (small).** Have the no-CGO `ParseFile` return its metadata with a `nil` error so files are still indexed, just without symbols. The repo map degrades to a file listing — genuinely useful context — instead of vanishing. Surface the degraded state in `baryo doctor`, which currently reports nothing about CGO or tree-sitter.

**2. Ship real symbols.** Build release artifacts with CGO enabled. `zig cc` is the cleanest cross-compiler here: one toolchain builds `linux/amd64` and `linux/arm64` against musl, producing static binaries with no glibc version floor, preserving the single-portable-binary property. macOS uses native clang on a macOS runner; Windows can stay no-CGO on the degraded path from step 1.

**3. Guard it.** Add a test that fails if a CGO build indexes zero files, and build the release binary in CI with the same flags the release uses.

## Acceptance criteria

- [ ] A no-CGO build indexes files and returns a non-empty repo map (symbols absent, paths present).
- [ ] `baryo doctor` reports whether tree-sitter symbol extraction is available.
- [ ] Released `linux/arm64` and `linux/amd64` binaries extract symbols.
- [ ] CI fails if a CGO build indexes zero files.
- [ ] CI builds the release binary with the same flags `goreleaser` uses.

### PR DESCRIPTION
**Pull request by arnelirobles**
The tree-sitter parsers are cgo-only, and the `CGO_ENABLED=0` stub returned an error alongside the file metadata. `Build` and `Update` treat an error as "unparseable" and skip the file, so every file was dropped. goreleaser builds releases with `CGO_ENABLED=0`, so **every published binary shipped an empty repo map** and the model got no view of the project at all.

The stub now returns a nil error. Missing symbols degrade the repo map to a file list, which is a real loss but a survivable one; an error removed the project entirely.

**Why it survived.** CI runs the suite twice, with and without CGO, but no test asserted that the index contains anything, so the nocgo pass had nothing to fail on. The new test is deliberately untagged and checks both file count and a non-empty repo map. Verified by restoring the error:

```
CGO_ENABLED=0 (bug restored)   FAIL  FileCount = 0, want 3
                                     RepoMap is empty
CGO_ENABLED=1 (bug restored)   ok
```

That asymmetry is how it shipped, and it is the same shape as the other gates in this repo that could not fail.

Also:

- `SymbolsAvailable` is a build-tagged constant, and `doctor` now reports symbol extraction instead of staying silent. It is a warning, which `AllPassed` treats as passing, and a test asserts that so a nocgo build still starts.
- CI builds through goreleaser (`build --snapshot --single-target`), so a broken release config is caught before a tag rather than by a release.
- First tests for `internal/doctor`.

Not in this PR, deliberately: restoring symbol extraction for released binaries. That is either a pure-Go tree-sitter runtime or a cgo release matrix, and it deserves a spike that can fail without blocking this fix. Until then the repo map in a release is paths only, and doctor now says so.

`gofmt`, `go vet`, and `go test -count=1 ./...` clean under both `CGO_ENABLED=0` and `CGO_ENABLED=1`.

Closes #9

## Summary by CodeRabbit

* **Bug Fixes**
  * Repository indexes now retain supported source files even when symbol extraction is unavailable.
  * Builds without CGO no longer omit files from repository maps; they provide file listings without symbol details.
  * Doctor diagnostics now clearly indicate whether symbol support is available and report limitations as non-blocking warnings.

* **Tests**
  * Added coverage for indexing behavior and symbol-support diagnostics across supported and limited build environments.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -71,6 +71,13 @@ jobs:
       - name: Test install.sh checksum verification
         run: sh scripts/install_test.sh
 
+      # Builds through the release path, so a broken .goreleaser.yaml or a
+      # release-only build flag is caught before a tag is pushed.
+      - uses: goreleaser/goreleaser-action@v6
```
[the remark this answers is not in the document]
Comment (arnelirobles) [reply .github/workflows/ci.yml]: Valid, and wider than this line: the repo had ten unpinned action uses, and the same goreleaser action in release.yml runs with contents: write while the step flagged here only has a read-only token. Pinning just this one would have implied a policy that was not enforced.

All ten are now pinned to commit SHAs across ci.yml, release.yml and self-assign.yml, with the tag kept in a trailing comment so upgrades stay readable. goreleaser-action resolves to e435ccd777264be153ace6237001ef4d979d3a7a, matching the revision named above.
